File: test_admin.py
import json, time, os
from playwright.sync_api import sync_playwright

BASE = 'http://localhost:3000'
SCREENSHOTS = '/tmp/admin_test'
os.makedirs(SCREENSHOTS, exist_ok=True)

# Credenciales admin — ajustar si son distintas
ADMIN_EMAIL = 'admin@admin.com'
ADMIN_PASS = 'admin123'

def shot(page, name):
    path = f'{SCREENSHOTS}/{name}.png'
    page.screenshot(path=path, full_page=True)
    print(f'  [screenshot] {path}')

def log(msg):
    print(f'>> {msg}')

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page()
    errors = []
    page.on('console', lambda m: errors.append(f'[{m.type}] {m.text}') if m.type == 'error' else None)

    # 1. Home
    log('Navegando a home...')
    page.goto(BASE)
    page.wait_for_load_state('networkidle')
    shot(page, '01_home')

    # 2. Login
    log('Abriendo modal de login...')
    try:
        page.click('text=Ingresar', timeout=5000)
    except:
        page.click('[data-section="login"], #btnLogin, .btn-login', timeout=5000)
    page.wait_for_load_state('networkidle')
    shot(page, '02_login_modal')

    log('Completando credenciales...')
    # Buscar inputs de email y password
    page.fill('input[type="email"], input[name="email"], #loginEmail', ADMIN_EMAIL)
    page.fill('input[type="password"], input[name="password"], #loginPassword', ADMIN_PASS)
    shot(page, '03_login_filled')

    # Submit
    page.click('button[type="submit"], #btnLoginSubmit, .btn-login-submit', timeout=5000)
    page.wait_for_load_state('networkidle')
    time.sleep(1)
    shot(page, '04_after_login')

    # Verificar login exitoso
    content = page.content()
    if 'admin' in content.lower() or 'panel' in content.lower():
        log('Login exitoso (detectado texto admin/panel)')
    else:
        log('ADVERTENCIA: No se detectó sesión admin claramente')

    # 3. Navegar a panel admin
    log('Buscando enlace al panel admin...')
    try:
        page.click('text=Admin, text=Panel Admin, [data-section="admin"]', timeout=5000)
        page.wait_for_load_state('networkidle')
        time.sleep(1)
        shot(page, '05_admin_panel')
        log('Panel admin cargado')
    except Exception as e:
        log(f'No se encontró enlace admin directo: {e}')
        # Intentar navegación directa por JS
        page.evaluate("showSection && showSection('admin')")
        time.sleep(1)
        shot(page, '05_admin_panel_js')

    # 4. Stats admin
    log('Revisando stats...')
    try:
        page.click('text=Estadísticas, text=Stats, [data-tab="stats"]', timeout=3000)
        time.sleep(1)
        shot(page, '06_admin_stats')
    except:
        log('No se encontró tab de stats (puede estar inline)')

    # 5. Usuarios admin
    log('Revisando usuarios...')
    try:
        page.click('text=Usuarios, [data-tab="users"]', timeout=3000)
        time.sleep(1)
        shot(page, '07_admin_users')
    except:
        log('No se encontró tab de usuarios')

    # 6. Vehículos admin
    log('Revisando vehículos...')
    try:
        page.click('text=Vehículos, [data-tab="vehicles"]', timeout=3000)
        time.sleep(1)
        shot(page, '08_admin_vehicles')
    except:
        log('No se encontró tab de vehículos')

    # 7. Reportes admin
    log('Revisando reportes...')
    try:
        page.click('text=Reportes, [data-tab="reports"]', timeout=3000)
        time.sleep(1)
        shot(page, '09_admin_reports')
    except:
        log('No se encontró tab de reportes')

    # 8. Sección de vehículos pública
    log('Revisando listado de vehículos...')
    try:
        page.click('text=Vehículos, nav text=Vehículos', timeout=3000)
        page.wait_for_load_state('networkidle')
        time.sleep(1)
        shot(page, '10_vehicles_list')
    except Exception as e:
        log(f'Error en listado: {e}')

    # 9. Mis vehículos
    log('Revisando mis vehículos...')
    try:
        page.click('text=Mis Vehículos, text=Mis vehículos', timeout=3000)
        page.wait_for_load_state('networkidle')
        time.sleep(1)
        shot(page, '11_my_vehicles')
    except Exception as e:
        log(f'Error en mis vehículos: {e}')

    # 10. Notificaciones
    log('Revisando notificaciones...')
    try:
        page.click('text=Notificaciones, #btnNotifications, .btn-notifications', timeout=3000)
        page.wait_for_load_state('networkidle')
        time.sleep(1)
        shot(page, '12_notifications')
    except Exception as e:
        log(f'Error en notificaciones: {e}')

    # Reporte final de errores JS
    log(f'\n=== ERRORES JS DETECTADOS ({len(errors)}) ===')
    for e in errors:
        print(f'  {e}')

    browser.close()
    log('\nTest completado. Screenshots en: ' + SCREENSHOTS)

File: test_admin.mjs
import { chromium } from 'playwright';
import fs from 'fs';
import path from 'path';

const BASE = 'http://localhost:3000';
const SCREENSHOTS = 'C:/tmp/admin_test';
fs.mkdirSync(SCREENSHOTS, { recursive: true });

const ADMIN_EMAIL = 'admin@admin.com';
const ADMIN_PASS = 'admin123';

const browser = await chromium.launch({ headless: true });
const context = await browser.newContext();
const page = await context.newPage();
const jsErrors = [];
page.on('console', msg => { if (msg.type() === 'error') jsErrors.push(msg.text()); });
page.on('pageerror', err => jsErrors.push('PAGE ERROR: ' + err.message));

const shot = async (name) => {
  const p = path.join(SCREENSHOTS, `${name}.png`);
  await page.screenshot({ path: p, fullPage: true });
  console.log(`  [📸] ${p}`);
};
const log = msg => console.log(`>> ${msg}`);

// ─── 1. HOME ─────────────────────────────────────────────────────────────────
log('=== 1. HOME ===');
await page.goto(BASE);
await page.waitForLoadState('networkidle');
await shot('01_home');

// ─── 2. LOGIN ────────────────────────────────────────────────────────────────
log('=== 2. LOGIN ===');
// Login es una sección, no un modal
await page.evaluate(() => showSection('login'));
await page.waitForTimeout(600);
await shot('02_login_section');

// Llenar y enviar
await page.fill('#loginEmail', ADMIN_EMAIL);
await page.fill('#loginPassword', ADMIN_PASS);
await shot('03_credentials');
await page.click('#loginBtn, #login button[type="submit"]');
await page.waitForLoadState('networkidle');
await page.waitForTimeout(2000);
await shot('04_post_login');

// Verificar si "Admin" es visible en navbar
const adminNavVisible = await page.locator('#navAdmin').isVisible();
log(`Admin en navbar visible: ${adminNavVisible}`);

if (!adminNavVisible) {
  log('ADVERTENCIA: Usuario no es admin o login falló');
  // Mostrar el toast si hay
  const toast = await page.locator('.toast, .alert, .notification').first().textContent().catch(() => '');
  log(`Toast/mensaje: ${toast}`);
}

// ─── 3. PANEL ADMIN ──────────────────────────────────────────────────────────
log('=== 3. PANEL ADMIN ===');
await page.evaluate(() => showSection('admin'));
await page.waitForLoadState('networkidle');
await page.waitForTimeout(1500);
await shot('05_admin_panel');

// Stats del admin
const statsHtml = await page.locator('#adminStats').innerHTML().catch(() => '(vacío)');
log(`adminStats HTML (100c): ${statsHtml.slice(0, 100)}`);

// ─── 4. TABS ADMIN ───────────────────────────────────────────────────────────
log('=== 4. TABS ADMIN ===');
const tabs = await page.locator('.admin-tab').allTextContents();
log(`Tabs encontrados: ${tabs.join(' | ')}`);

for (const tab of tabs) {
  log(`  Clickeando tab: ${tab}`);
  await page.click(`.admin-tab:has-text("${tab}")`);
  await page.waitForTimeout(1200);
  const safeName = tab.toLowerCase()
    .replace(/á/g,'a').replace(/é/g,'e').replace(/í/g,'i').replace(/ó/g,'o').replace(/ú/g,'u')
    .replace(/[^a-z0-9]/g,'_');
  await shot(`06_admin_tab_${safeName}`);

  // Contar filas en tabla
  const rows = await page.locator('#adminContent table tbody tr, #adminContent .list-item').count();
  log(`  Filas/items en "${tab}": ${rows}`);
}

// ─── 5. VEHÍCULOS PÚBLICOS ───────────────────────────────────────────────────
log('=== 5. VEHÍCULOS ===');
await page.evaluate(() => showSection('vehicles'));
await page.waitForLoadState('networkidle');
await page.waitForTimeout(1500);
await shot('07_vehicles_list');
const vehicleCards = await page.locator('.vehicle-card, .card, [class*="vehicle"]').count();
log(`Cards de vehículos: ${vehicleCards}`);

// ─── 6. DETALLE DE VEHÍCULO ──────────────────────────────────────────────────
log('=== 6. DETALLE VEHÍCULO ===');
try {
  await page.locator('.vehicle-card, .card').first().click({ timeout: 3000 });
  await page.waitForLoadState('networkidle');
  await page.waitForTimeout(1500);
  await shot('08_vehicle_detail');

  // Verificar mapa
  const mapVisible = await page.locator('#vehicleMap, .leaflet-container').isVisible();
  log(`Mapa visible: ${mapVisible}`);

  // Verificar historial de precios
  const priceHistory = await page.locator('.price-history, #priceHistoryChart').isVisible().catch(() => false);
  log(`Historial precios visible: ${priceHistory}`);

  // Verificar botón compartir
  const shareBtn = await page.locator('[onclick*="shareVehicle"], .share-btn, text=Compartir').isVisible().catch(() => false);
  log(`Botón compartir visible: ${shareBtn}`);
} catch (e) {
  log(`No se pudo abrir detalle: ${e.message}`);
}

// ─── 7. PUBLICAR ─────────────────────────────────────────────────────────────
log('=== 7. PUBLICAR ===');
await page.evaluate(() => showSection('publish'));
await page.waitForTimeout(1000);
await shot('09_publish_form');
const publishInputs = await page.locator('#publish input, #publish select, #publish textarea').count();
log(`Inputs en formulario publicar: ${publishInputs}`);

// ─── 8. MIS ANUNCIOS ─────────────────────────────────────────────────────────
log('=== 8. MIS ANUNCIOS ===');
await page.evaluate(() => showSection('my-vehicles'));
await page.waitForLoadState('networkidle');
await page.waitForTimeout(1500);
await shot('10_my_vehicles');

// ─── 9. MENSAJES ─────────────────────────────────────────────────────────────
log('=== 9. MENSAJES ===');
await page.evaluate(() => showSection('messages'));
await page.waitForLoadState('networkidle');
await page.waitForTimeout(1500);
await shot('11_messages');

// ─── 10. PERFIL ──────────────────────────────────────────────────────────────
log('=== 10. PERFIL ===');
await page.evaluate(() => showSection('profile'));
await page.waitForLoadState('networkidle');
await page.waitForTimeout(1500);
await shot('12_profile');

// ─── REPORTE FINAL ───────────────────────────────────────────────────────────
log('\n=== REPORTE DE ERRORES JS ===');
if (jsErrors.length === 0) {
  log('✅ Sin errores JS detectados');
} else {
  log(`❌ ${jsErrors.length} error(es):`);
  jsErrors.forEach((e, i) => log(`  ${i+1}. ${e}`));
}

await browser.close();
log(`\n✅ Test completo. Screenshots en: ${SCREENSHOTS}`);
